### DISCUSSION TITLE
Re-useability of WhereClause

### DISCUSSION THREAD

**Discussion by bhavinbusa** [cdisc-org/analysis-results-standard#95] [General]
Hi team, let's use this discussion page on the re-useability of WhereClause. Do we want to add an ID to the WhereClause?

**Comment by ASL-rmarshall** [comment]
Note that there are already some comments in issue #15

**Comment by ASL-rmarshall** [comment]
At the moment (as of end of Sprint 11), compound expressions are specified differently for different types (specializations) of the `WhereClause` class:
- For `AnalysisSet` and `Group` (which includes both `AnalysisGroup` and `DataGroup`), compound expressions can only be defined with reference to other identified `AnalysisSet`s or `Group`s - each of the sub-clauses (`whereClauses`) in the 'CompoundExpression` is a reference to the `id` of an existing `AnalysisSet` or `Group` (respectively).
- For `DataSubset`, compound expressions can only be defined as combinations of fully defined sub-clauses (`whereClauses`), each of which is either a `Condition` or another `CompoundExpression`).

I think that the model should be updated to accommodate either approach for any type of where clause (i.e., each sub-clause either can be a reference to an existing, identified type of where clause or can be fully specified).
